pkg/riff/commands/application_status.go

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package commands
+
+import (
+	"context"
+	"fmt"
+	"github.com/projectriff/riff/pkg/cli"
+	buildv1alpha1 "github.com/projectriff/system/pkg/apis/build/v1alpha1"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strings"
+	"time"
+)
+
+type ApplicationStatusOptions struct {
+	cli.ResourceOptions
+	Since string
+}
+
+var (
+	_ cli.Validatable = (*ApplicationStatusOptions)(nil)
+	_ cli.Executable  = (*ApplicationStatusOptions)(nil)
+)
+
+func (opts *ApplicationStatusOptions) Validate(ctx context.Context) *cli.FieldError {
+	errs := cli.EmptyFieldError
+
+	errs = errs.Also(opts.ResourceOptions.Validate(ctx))
+
+	return errs
+}
+
+func (opts *ApplicationStatusOptions) Exec(ctx context.Context, c *cli.Config) error {
+	app, err := c.Build().Applications(opts.Namespace).Get(opts.Name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	now := time.Now()
+	info := make([]map[string]string, 5)
+	info[0] = map[string]string{"Name": app.Name}
+	info[1] = map[string]string{"Namespace": app.Namespace}
+	info[2] = map[string]string{"Status": fmt.Sprintf("%s",
+		cli.FormatConditionStatus(app.Status.GetCondition(buildv1alpha1.FunctionConditionReady)))}
+	info[3] = map[string]string{"Latest Image": fmt.Sprintf("%s",
+		cli.FormatEmptyString(app.Status.LatestImage))}
+	info[4] = map[string]string{"Age": fmt.Sprintf("%s",
+		cli.FormatTimestampSince(app.CreationTimestamp, now))}
+	printStatusMessage(c, "Application", info)
+	cond := app.Status.GetCondition(buildv1alpha1.FunctionConditionReady)
+	if cond != nil {
+		printStatusMessage(c, "Build Status", buildConditionStatusInfo(cond))
```

Comment:
I'm confused by the name "Build Status", this using the ready condition for the application. There is a [more specific condition](https://github.com/projectriff/system/blob/a3d889fa56cd8efe1156f702fefcd8ec88326c26/pkg/apis/build/v1alpha1/application_lifecycle.go#L30) for the status of the build.

Reply:
changing this to "Status Info"

Reply:
It can be either FunctionConditionReady or FunctionConditionBuildSucceeded depending on what failed